### PR TITLE
Favorite Servers

### PR DESCRIPTION
**Pull request by theunknownxy**
This is a implementation of #436.

TODO
- [x] How to mark favorite servers?
- [x] How to render favorite servers?

### CODE REVIEW

**Comment by theunknownxy** [comment]
Tinting the background of favorite servers looks pretty decent.
Highlighting
Any opinions for the color?

**Comment by VelocityRa** [comment]
Took me a good 10 seconds to notice it, make it more prominent!

**Comment by feikname** [comment]
How about keeping the blue background and changing text color to yellow/gold? Yellow and blue go good together!

 The same color that @VelocityRa used in the Spectator list sounds good to me.
image

**Comment by theunknownxy** [comment]
Here you go:
Yellow!

**Comment by noway421** [comment]
Just say when it's ready as it is now marked as [WIP]. Looks really good.

**Comment by theunknownxy** [comment]
If nobody has any suggestions it's ready. 
Also you can't change labels of your issues if you aren't a collaborator. What about removing the "work-in-progress" label and instead use the convention that prefixing a PR with "WIP: " means it's work in progress?

**Comment by feikname** [comment]
Me and @VelocityRa log in almost everyday to ensure tagging is correct. 

But I have to agree, it's a pain in the ass for people wanting to submit a pull request that they can't change their own tags. The suggestion of modifying the title (or just the contents, really, it's rare to have more than 4 issues pull requests open and it's fast to open and see their status) sounds good to me.

@VelocityRa should we remove the `PR: work-in-progress` label?

File: Sources/Gui/MainScreenHelper.h [hunk 1]
```diff
@@ -59,6 +61,8 @@ namespace spades {
 			int GetPing() { return ping; }
 			int GetNumPlayers() { return numPlayers; }
 			int GetMaxPlayers() { return maxPlayers; }
+			bool IsFavorite() { return favorite; }
+			bool SetFavorite(bool favorite) { this->favorite = favorite; }
```
Comment (yvt) [inline Sources/Gui/MainScreenHelper.h]: `return` is missing in this non-void function. (specified by C++ spec as undefined behavior)

Comment (theunknownxy) [inline Sources/Gui/MainScreenHelper.h]: Oops it's not only undefined behavior, it also makes no sense.

**Comment by VelocityRa** [comment]
@feikname @theunknownxy **WIP:** sounds good to me. Let's ditch the "PR:work-in-progress" label.

**Comment by feikname** [comment]
@VelocityRa done.

**Comment by yvt** [comment]
Thank you for fix. Looks very good.

**Pull request merged**